src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -47,23 +54,42 @@ public String send(Letter letter, String serviceName) throws JsonProcessingExcep
 
         log.info("Generated message: id = {} for letter with print queue id = {} ", messageId, letter.type);
 
-        sendClient
-            .sendAsync(createQueueMessage(new WithServiceName<>(letter, serviceName), messageId))
-            .thenRunAsync(() -> {
-                log.info("Message acknowledged: id = {}", messageId);
-            })
-            .thenRunAsync(sendClient::closeAsync);
+        Message message = createQueueMessage(new WithServiceName<>(letter, serviceName), messageId);
+        Instant startSending = Instant.now();
+
+        CompletableFuture<Void> sendResult = sendClient
+            .sendAsync(message)
+            .whenCompleteAsync((result, exc) -> {
+                Duration tookSending = Duration.between(startSending, Instant.now());
+                insights.trackMessageAcknowledgement(tookSending, exc != null, messageId);
+
+                if (exc != null) {
```

Comment:
you could extract the null check to a variable as it is getting repeated `boolean isException=exc != null;`

Reply:
Also you could refactor the code as below
```
 CompletableFuture<Void> sendResult = sendClient
            .sendAsync(message)
            .whenCompleteAsync(onCompleteAsync(messageId, startSending))
            .thenRunAsync(sendClient::closeAsync);
```

Reply:
no it cannot be refactored as we specifically want to capture success/failure of message send event, which is the first call to the client